doc/conf.py

```diff
@@ -325,6 +327,10 @@ def setup(app):
         'numpy': 'http://docs.scipy.org/doc/numpy-1.9.1'},
     'examples_dirs': examples_dirs,
     'gallery_dirs': gallery_dirs,
+    'subsection_order': ExplicitOrderStrict(['../examples/sin_func',
```

Comment:
It's not exactly clear to me from this PR how this `ExplicitOrderStrict` class is of benefit to a user over just specifying a list?